internal/db/schema/migrations/oss/postgres/48/03_create_kms_key_destructions.up.sql

```diff
@@ -0,0 +1,40 @@
+begin;
+
+  create table kms_key_destruction_status_enm (
+    name text primary key
+      constraint only_predefined_values_allowed
+      check(name in ('pending', 'running', 'completed', 'failed'))
+  );
+  comment on table kms_key_destruction_status_enm is
+    'Table holding valid statuses a key destruction can take';
+
+  insert into kms_key_destruction_status_enm (name)
+  values
+    ('pending'),
+    ('running'),
+    ('completed'),
+    ('failed');
+
+  create trigger 
+    immutable_columns
+  before
+  update on kms_key_destruction_status_enm
+    for each row execute procedure immutable_columns('name');
+
+  create table kms_key_destruction (
+    private_id wt_private_id primary key,
+    key_id wt_private_id not null, -- Note: not a foreign key, outlives original reference
+    activation_time wt_timestamp not null,
+    deactivation_time wt_timestamp not null,
+    destruction_start_time wt_timestamp not null,
+    destruction_end_time wt_timestamp null default null,
+    status text not null
```

Comment:
* What does it mean if the `status` is `completed` and the `destruction_end_time` is null? 
* What does it mean if the `status` is `pending`, `running`, or `failed` and the `destruction_end_time` is not null?

Reply:
> What does it mean if the status is `completed` and the destruction_end_time is null?

This is an invalid state. Do you know any way we could enforce this in the schema?

> What does it mean if the status is `pending`, `running`, or `failed` and the destruction_end_time is not null?

For statuses `pending`, and `running` this is also an invalid state. For `failed`, this would be expected to be set (same as for `completed`).